makefile

```diff
@@ -33,41 +40,78 @@ override DMD_FLAGS += $(DFLAGS)
 override LDC_FLAGS += $(DFLAGS)
 override GDC_FLAGS += $(DFLAGS)
 DMD_TEST_FLAGS = -w -g -Jbin -version=StdLoggerDisableWarning
-override LDC_FLAGS += -O5 -release -oq -d-version=StdLoggerDisableWarning
-override GDC_FLAGS += -O3 -frelease -d-version=StdLoggerDisableWarning
+override LDC_FLAGS += -O5 -release -oq -d-version=StdLoggerDisableWarning -Jbin
+override GDC_FLAGS += -O3 -frelease -d-version=StdLoggerDisableWarning -Jbin
 SHELL:=/usr/bin/env bash
 
-all: dmdbuild
-ldc: ldcbuild
-gdc: gdcbuild
+DMD_BIN = bin/dmd/dscanner
+LDC_BIN = bin/ldc/dscanner
+GDC_BIN = bin/gdc/dscanner
```

Comment:
the executable should not be grouped by compiler, also previous tests depend on it being in bin/dscanner as well as this no longer matching dub builds.

Reply:
The reason I grouped by compiler was to ensure that linked object files were all compiled by the same compiler.
I fear that otherwise (linking objects compiled by different compilers) might lead to UB.

Reply:
you can group object files by compiler, but the resulting exe path must not change